test/integration/targets/docker_node/tasks/test_node.yml

```diff
@@ -0,0 +1,309 @@
+---
+- block:
+  - name: Make sure we're not already using Docker swarm
+    docker_swarm:
+      state: absent
+      force: true
+
+  - name: Try to get docker_node_facts when docker is not running in swarm mode
+    docker_node_facts:
+    ignore_errors: yes
+    register: output
+
+  - name: assert failure when called when swarm is not in use or not run on manager node
+    assert:
+      that:
+         - 'output is failed'
+         - 'output.msg == "Error running docker swarm module: must run on swarm manager node"'
+
+  - name: Create a Swarm cluster
+    docker_swarm:
+      state: present
+    register: output
+
+  - name: assert changed when create a new swarm cluster
+    assert:
+      that:
+         - 'output is changed'
+         - 'output.actions[0] | regex_search("New Swarm cluster created: ")'
+         - 'output.swarm_facts.JoinTokens.Manager'
+         - 'output.swarm_facts.JoinTokens.Worker'
+
+  - name: Try to get docker_node_facts when docker is running in swarm mode and as manager
+    docker_node_facts:
+    register: output
+
+  - name: assert reading docker swarm node facts
+    assert:
+      that:
+         - 'output.nodes_facts | length > 0'
+         - 'output.nodes_facts[0].ID is string'
+
+  - name: Register node ID
+    set_fact:
+      nodeid: "{{ output.nodes_facts[0].ID }}"
+
+  - name: Try to set node as manager
+    docker_node:
+      hostname: "{{ nodeid }}"
+      role: manager
+    register: output
+
+  - name: assert that node role does has not changed
+    assert:
+      that:
+         - 'output is not changed'
+         - 'output.node_facts.Spec.Role == "manager"'
+
+  - name: Try to set node as worker
+    docker_node:
+      hostname: "{{ nodeid }}"
+      role: worker
+    ignore_errors: yes
+    register: output
+
+  - name: assert that node cannot change role to worker
+    assert:
+      that:
+         - 'output is failed'
+         - 'output.msg | regex_search("attempting to demote the last manager of the swarm")'
+
+  - name: Try to set node availability as paused
+    docker_node:
+      hostname: "{{ nodeid }}"
+      availability: pause
```

Comment:
You should do this another time, to see whether this operation is idempotent (same as for the ones below).

Reply:
Also, since the module supports check mode, the common pattern should be:
1. Do change in check mode
2. Do change
3. Do change again (idempotent)
4. Do change again (idempotent, check mode)
5. Assert that
   - 1 is changed
   - 2 is changed
   - 3 is not changed
   - 4 is not changed

See [here](https://github.com/ansible/ansible/blob/devel/test/integration/targets/docker_container/tasks/tests/start-stop.yml#L13-L52) for an example. (Most docker_* tests don't do that yet, but that's another thing which should be changed eventually.)

Reply:
Changes made